### PR TITLE
Programming Languages - NodeJs Complete Resources

### LINKED ISSUE #442: 💡 [FEATURE] - Add "NodeJs" Complete Resources
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

Node.js, a powerful JavaScript runtime built on Chrome's V8 JavaScript engine, is widely used for building scalable network applications. Despite its popularity and extensive use in backend development, the current documentation lacks comprehensive resources dedicated to Node.js.

### Explain your solution

I propose adding a dedicated section for Node.js resources in web development, such as

```
./Web Development
    /Web Frameworks
         /NodeJs
            /readme.md

```

which will include:

- Node.js Basics:
    - Setting up the Node.js environment
    - Understanding the event-driven architecture
    - Core modules and the Node.js standard library
- Advanced Node.js:
    - Asynchronous programming and Promises
    - Building RESTful APIs
    - Using Express.js for web development
    - Working with databases (MongoDB, SQL)
- Node.js Tools and Libraries:
    - Package management with npm and yarn
    - Popular frameworks and libraries (Express, Koa, Hapi)
    - Testing with Mocha and Chai
- Performance and Optimization:
    - Profiling and debugging Node.js applications
    - Best practices for scalability
- Deployment:
    - Deployment strategies for Node.js applications
    - Using Docker with Node.js
    - Continuous integration and delivery
- Node.js Projects:
    - Real-world applications

### Any alternative approaches/features

To enhance the learning experience, I suggest including essential resources such as recommended books, YouTube channels, and websites. Additionally, providing a beginner-friendly roadmap can help new developers navigate through the resources step-by-step.

### Additional Context

This feature request aims to enrich the Web development section by providing extensive learning materials for Node.js, similar to existing resources for other technologies. This will cater to the high demand for Node.js knowledge in backend development and improve the overall utility of our documentation for developers at all levels.

### PR DESCRIPTION
**Pull request by Santhosh-Siddhardha**
**IMPORTANT:** When creating your Pull Request, please use the following format for the title: `Category - Resource Name & Section`  

PR Title samples:

- `Web development - CSS Portal & Tools`
- `Web development - Leetcode & Challenges`

Closes #442 

### Description: 
Hey @jfmartinz I have added the resources of NodeJs

```
 ./Programming Languages
    /NodeJs
     /readme.md
```
Please take a look, Thank you.

### Checklist:
Before submitting your pull request, ensure that you have completed the following tasks: 

- [x] I have carefully reviewed and adhered to the [contributing guidelines](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) before creating this pull request.
- [x] I followed the prescribed PR title template. _(Check this if you're adding a resource)_

> fill the bracket with x. Something like this [x]

### CODE REVIEW

File: Programming Languages/NodeJs/readme.md [hunk 1]
```diff
@@ -0,0 +1,1167 @@
...
+need for multiple programming languages.
+- **Rich Ecosystem**: Node.js's package manager, npm, provides access to a vast repository of open-source libraries and modules, facilitating rapid development and the 
+addition of new functionalities.
+- **Strong Community Support**: The Node.js community is vibrant and growing, offering a wealth of resources, tutorials, and forums where developers can collaborate and 
+share knowledge.
+
+Node.js is widely used for building various types of applications, including web servers, RESTful APIs, real-time chat applications, and microservices architectures. 
+Its efficiency, versatility, and robust ecosystem make Node.js a popular choice for modern web development.
+
+These resources are designed to help you build a strong foundation in Node.js and apply it effectively to your server-side and real-time application development tasks. Whether you are a beginner starting with the basics or an experienced developer looking to deepen your understanding, the following sections cover various aspects of Node.js, including tutorials, coding platforms, tools, and more.
```
Comment (jfmartinz) [inline Programming Languages/NodeJs/readme.md:22]:
```suggestion

```

please remove key features

**Review by jfmartinz — CHANGES_REQUESTED** [review]

File: Programming Languages/NodeJs/readme.md (hunk 1 above)
Comment (Santhosh-Siddhardha) [reply Programming Languages/NodeJs/readme.md]: Removed Key features

**Comment by Santhosh-Siddhardha** [comment]
@jfmartinz Made changes, Have a look

**Review by jfmartinz — APPROVED** [review]
Thanks

**Comment by Santhosh-Siddhardha** [comment]
@jfmartinz Please assign this PR to me.

**Pull request merged**